packages/rocketchat-language-plugin/server/add_or_update_user.js

```diff
@@ -0,0 +1,26 @@
+Meteor.methods({
+	add_or_update_user(channel, plugin_name) {
+		const channel_exist = Meteor.users.findOne({
+			$and: [{ _id: Meteor.userId()}, {'automatic_channels.name': channel}]
+		});
+		if (!channel_exist) {
+			Meteor.users.update(Meteor.userId(), {
+				$push: {
+					automatic_channels: {'name': channel, 'plugin': plugin_name, 'added_to':'true', 'blacklisted': 'false'}
+				}
+			});
+		} else {
+			Meteor.users.update({
+				_id: Meteor.userId(),
+				'automatic_channels.name': channel
+			}, {
+				$set: {
+					'automatic_channels.$.added_to' :'true'
```

Comment:
But Consider the case - 
1. For eg -if a user was added to English initially.
2. Then he left the channel, so the blacklisted was set to true.
3.Then he changed his language to French, and subdocument (the one which had English) was automatically removed because we do not want the user to remain in more than one language channel.
4.Now user changed his language again to English(which was blacklisted by him).So now we do not have any information that English was blacklisted by him because we already removed the subdocument.
Is that ok for us?